### ISSUE TITLE
Release new version

### ISSUE THREAD

**Issue by MikeMcQuaid** [mas-cli/mas#512]
**Is your feature request related to a problem? Please describe.**
https://github.com/mas-cli/mas/pull/496 was merged but no stable release contains the fix for `mas outdated` so tools like `brew bundle` still cannot rely on the output.

**Describe the solution you'd like**
A new `mas` release.

**Describe alternatives you've considered**
Wait patiently.

**Additional context**
See https://github.com/Homebrew/homebrew-bundle/pull/1308 for the homebrew-bundle PR blocked on a new release.

Thanks for the great work on this project, I personally rely on it and love it! ❤️

**Comment by wzzrd** [comment]
So ehm, I can't write a single line of Swift or Objective C, but if there's anything I can do to push this forward, I would be more than happy to. It seems the release PR is hanging on some jobs not having run, is it understood why that's not happening?

**Comment by rgoldberg** [comment]
The project lead, @phatblat, has been sporadically working on betas for 1.8.7.

1.8.7 includes some fixes for `mas` itself, plus many updates to the build, testing, GitHub actions / workflows, and release scripts.

I'm now triaging & deduplicating existing PRs & issues, and I'm starting to work on some bug fixes.

Hopefully, @phatblat will return soon to finish 1.8.7, review & merge additional bug fixes, document all the release & admin procedures, prioritize outstanding issues, and decide amongst solutions for various issues.

Someone will update when people have gotten in touch & we all know more about the current & future status of mas.

**Comment by rgoldberg** [comment]
I think that getting a stable release out should be almost as easy as getting an official release out. The build scripts, release scripts, etc., just need to be finished, reviewed & merged.

**Comment by MikeMcQuaid** [comment]
Sounds great! Note that Homebrew will only update to a stable release, not a beta one, in case that affects anything.

**Comment by juanlazarde** [comment]
Hello, unsure where to post this. I’ve **installed the latest beta version of mas-cli** using:

```bash
curl -L -o mas.pkg https://github.com/mas-cli/mas/releases/download/v1.8.7-beta.35/mas.pkg
sudo installer -pkg mas.pkg -target /
```

After installation, the `mas version` confirms as **1.8.7-beta.35**.

But, when I run `mas outdated`, I get **Error: Search failed**. I can install new packages (e.g., `mas install <some package>`), but `mas upgrade` doesn’t work.

Version `1.8.6` was working fine, except I couldn't upgrade certain apps like "Canva" and "CapCut," which showed as outdated but didn't update, over an over, hence the update to a beta.

Could you please guide me on the **proper installation method** and troubleshooting steps for these issues? Thank you!

**Comment by rgoldberg** [comment]
@juanlazarde

Thanks for the heads up about this issue.

You should use the code from the `main` branch in this mas GitHub repo instead of any of the betas, because the code in `main` has many improvements that are not in the beta. (see the last section of this comment to learn about this)

# 2 options to use `main` branch from GitHub
## Option 1: Homebrew
If you use Homebrew, you can install what is in `main` via the following command line:

```bash
brew install --HEAD mas
```

That will build mas from GitHub `main` instead of installing pre-built binaries. Building `main` requires Xcode >= 13.3 & macOS >= 12.

## Option 2: Manual build of clone of mas GitHub repo
Using Homebrew (Option 1) would be much simpler than this Option 2.

Homebrew is also very handy in everyday usage, having nothing to do with helping you install `main` mas. You probably should look into using Homebrew if you aren't using it already.

If you can't/won't use Homebrew, you can clone the repo & build manually.

I think the procedure for setting up a new clone of the repo (after the repo has been cloned; all commands run in the root of the repo) then building the executable is:

```bash
$ ./script/bootstrap
$ ./script/build
# I haven't run the following script
$ ./script/install
```

The `install` script above should put the newly built `mas` in your path, but I haven't used it as I prefer using built executables from inside the repo since I'm rebuilding frequently. If you want to use `mas` builds from your repo, you can run:

```bash
/path/to/mas/repo/clone/.build/release/mas
```

Obviously, replace `/path/to/mas/repo/clone` in the above with the correct path to your clone of the mas repo.

# Notes about beta vs. `main`
I'm working on the `main` branch in GitHub; @phatblat is working on the betas. If I can confirm the problem in the newest beta & fix it easily, I'll let @phatblat know.

None of the betas are as good as what is in `main`, because `main` has many fixes that haven't been rebased into the betas. @phatblat probably should rebase the beta on `main`. I'll see if that is easy, then let him know.

**Comment by rgoldberg** [comment]
I'll provide more troubleshooting steps if the problem exists for you in `main`, but here is useful info for you to report if the problem occurs for you in `main`:

- macOS version
- Xcode version
- Mac model & architecture
- shell & version
- what apps are reported as outdated in the Mac App Store
- output of `mas list`

Not all of the above are strictly necessary, but since GitHub communication is asynchronous, having a dump of too much info is more helpful than too little.

`mas outdated` works fine for me from `main`. I'm not upgrading anything now to facilitate testing, but, last I tried, that worked in then-current `main`, too.

**Comment by rgoldberg** [comment]
@juanlazarde Also, normally it's best to open a bug report issue rather than jumping into an existing issue, unless the issue is exactly the same, but this isn't. I don't want to force you to make a new issue, but if this isn't resolved quickly, it would be helpful to have a separate issue.

**Comment by rgoldberg** [comment]
@juanlazarde `mas outdated` from the executable installed from the 1.8.7 beta 35 `mas.pkg` that you downloaded worked fine for me.

I installed it by double clicking on `mas.pkg` instead of running `sudo` (I didn't want to mess with what user owns the executable).

The executable compiled from the `v1.8.7-beta.35` branch also worked for me. So there's something wrong with it and your system, not just in general.

Do you use the App Store for a country other than the US?

**Comment by rgoldberg** [comment]
1.8.7 was released.

It is available via Homebrew & via [GitHub Releases](https://github.com/mas-cli/mas/releases/tag/v1.8.7).

It is not currently available via MacPorts.

**Comment by MikeMcQuaid** [comment]
Thanks @rgoldberg!

@phatblat should https://github.com/mas-cli/mas/pull/506 be merged considering this? I'm subscribed to that thread and found it a little confusing that it hasn't been.
